WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -136,13 +136,13 @@ private void changeLanguage(String languageCode) {
         Configuration conf = res.getConfiguration();
         // will return conf.locale if conf is non-null, or Locale.getDefault()
         Locale currentLocale = LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
+        Locale newLocale = WPPrefUtils.languageLocale(languageCode);
 
-        if (currentLocale.getLanguage().equals(new Locale(languageCode).getLanguage())) {
+        if (currentLocale.toString().equals(newLocale.getDisplayLanguage())) {
             return;
         }
 
-        Locale newLocale = WPPrefUtils.languageLocale(languageCode);
-        if (currentLocale.getLanguage().equals(newLocale.getLanguage())) {
+        if (Locale.getDefault().toString().equals(newLocale.toString())) {
```

Comment:
Yeah, that was intentional. As I was testing I switched between `en_US` and `en_GB` but I noticed that this comparison returned true. Turns out `getLanguage` doesn't include the country code so regional variants always compare to true.